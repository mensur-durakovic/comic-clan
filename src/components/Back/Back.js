import React from "react";
import PropTypes from 'prop-types';
import styled from "styled-components";
import BackIconSvg from "../../static/BackIcon.svg";

const BackWrapper = styled.div`
  height: 70px;
  box-sizing: border-box;
  position: relative;
  padding-top: 34px;
  padding-left: 38px;
`;

const BackArrow = styled.span`
  width: 10px;
  height: 10px;
  background-image: url(${BackIconSvg});
  background-repeat: no-repeat;
  background-size: contain;
  position: absolute;
  top: 40px;
  left: 30px;
  cursor: pointer;
`;

const BackDescription = styled.span`
  font-weight: 500;
  font-size: 16px;
  line-height: 16px;
  text-decoration-line: underline;
  color: #777777;
  cursor: pointer;
  margin-left: 5px;
`;

const Back = React.memo(props => {
  const { clickHandler, text } = props;
  return (
    <BackWrapper>
      <BackArrow />
      <BackDescription onClick={clickHandler}> {text} </BackDescription>
    </BackWrapper>
  );
});

Back.propTypes = {
  text: PropTypes.string.isRequired,
  clickHandler: PropTypes.func.isRequired,
};

export default Back;
