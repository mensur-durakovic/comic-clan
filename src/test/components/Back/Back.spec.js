import React from "react";
import renderer from "react-test-renderer";
import Back from "../../../components/Back/Back";

test("Back component", () => {
  const tree = renderer
    .create(<Back text="test" clickHandler={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
