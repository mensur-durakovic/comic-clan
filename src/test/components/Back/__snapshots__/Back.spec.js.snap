// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Back component 1`] = `
<div
  className="sc-AxjAm ebRsjh"
>
  <span
    className="sc-AxirZ fwUwCO"
  />
  <span
    className="sc-AxiKw ibpvHa"
    onClick={[Function]}
  >
     
    test
     
  </span>
</div>
`;
